File: src/lightnap-ng/src/app/core/backend-api/dtos/users/request/search-notifications-request-dto.ts
import { NotificationStatus, NotificationType, PagedRequestDto } from "@core";

/**
 * Interface representing a request to search for the user's notifications.
 * Extends the PaginationRequest interface to include pagination properties.
 */
export interface SearchNotificationsRequestDto extends PagedRequestDto {
  /**
   * The ID to filter notifications since. Returned notifications will be since this ID.
   * @type {number}
   */
  sinceId?: number;

  /**
   * The ID to filter notifications before. Returned notifications will be from prior to this ID.
   * @type {number}
   */
  priorToId?: number;

  /**
   * The status of the notifications to filter by.
   * @type {NotificationStatus}
   */
  status?: NotificationStatus;

  /**
   * The type of the notifications to filter by.
   * @type {NotificationType}
   */
  type?: NotificationType;
}

File: src/LightNap.Core/Notifications/Dto/Request/SearchNotificationsRequestDto.cs
using LightNap.Core.Api;
using LightNap.Core.Data.Entities;

namespace LightNap.Core.Notifications.Dto.Request
{
    /// <summary>
    /// Data transfer object for searching notifications with pagination support.
    /// </summary>
    public class SearchNotificationsRequestDto : PagedRequestDtoBase
    {
        /// <summary>
        /// Gets or sets the ID of the notification to start the search from.
        /// </summary>
        /// <value>The ID of the notification to start the search from.</value>
        public int? SinceId { get; set; }

        /// <summary>
        /// Gets or sets the ID of the notification to end the search at.
        /// </summary>
        /// <value>The ID of the notification to end the search at.</value>
        public int? PriorToId { get; set; }

        /// <summary>
        /// Gets or sets the status of the notifications to search for.
        /// </summary>
        /// <value>The status of the notifications to search for.</value>
        public NotificationStatus? Status { get; set; }

        /// <summary>
        /// Gets or sets the type of the notifications to search for.
        /// </summary>
        /// <value>The type of the notifications to search for.</value>
        public NotificationType? Type { get; set; }
    }
}